catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/criteria/contextual/ContextualEvaluator.java

```diff
@@ -197,47 +192,48 @@ public static Directory buildIndex(String fullDocument) throws IOException {
-        new IndexWriter(index, contextualAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED)) {
-      logTokens(indexWriter.getAnalyzer(), FIELD_NAME, fullDocument, "ContextualAnalyzer");
-
-      // Add the indexable text to the case-insensitive index writer, assigning it the
-      // "case-insensitive" field name
-      addDoc(indexWriter, FIELD_NAME, indexableText);
-    }
-    CaseSensitiveContextualAnalyzer caseSensitiveStandardAnalyzer =
-        new CaseSensitiveContextualAnalyzer(Version.LUCENE_30);
-
-    // Create a second IndexWriter using the custom case-sensitive StandardAnalyzer
-    // NOTE: set boolean to false to append the case-sensitive indexed text to the existing
-    // index (populated by first IndexWriter)
-    try (IndexWriter csIndexWriter =
-        new IndexWriter(
-            index, caseSensitiveStandardAnalyzer, false, IndexWriter.MaxFieldLength.UNLIMITED)) {
-
-      // Add the indexable text to the case-sensitive index writer, assigning it the
-      // "case-sensitive" field name
-      addDoc(csIndexWriter, CASE_SENSITIVE_FIELD_NAME, indexableText);
+    try (ContextualAnalyzer contextualAnalyzer = new ContextualAnalyzer(Version.LUCENE_30)) {
+
+      // 1. create the index
+      try (Directory index = new RAMDirectory();
+          // Create an IndexWriter using the case-insensitive StandardAnalyzer
+          // NOTE: the boolean arg in the IndexWriter constructor means to create a new index,
+          // overwriting any existing index
+          IndexWriter indexWriter =
+              new IndexWriter(
+                  index, contextualAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED);
+          CaseSensitiveContextualAnalyzer caseSensitiveStandardAnalyzer =
+              new CaseSensitiveContextualAnalyzer(Version.LUCENE_30);
+
+          // Create a second IndexWriter using the custom case-sensitive StandardAnalyzer
+          // NOTE: set boolean to false to append the case-sensitive indexed text to the existing
+          // index (populated by first IndexWriter)
+          IndexWriter csIndexWriter =
+              new IndexWriter(
+                  index,
+                  caseSensitiveStandardAnalyzer,
+                  false,
+                  IndexWriter.MaxFieldLength.UNLIMITED)) {
+        // Retrieve the text from the document that can be indexed using the specified XPath
+        // selectors
+        String indexableText = getIndexableText(fullDocument, xpathSelectors);
+
+        logTokens(indexWriter.getAnalyzer(), FIELD_NAME, fullDocument, "ContextualAnalyzer");
+
+        // Add the indexable text to the case-insensitive index writer, assigning it the
+        // "case-insensitive" field name
+        addDoc(indexWriter, FIELD_NAME, indexableText);
+
+        // Add the indexable text to the case-sensitive index writer, assigning it the
+        // "case-sensitive" field name
+        addDoc(csIndexWriter, CASE_SENSITIVE_FIELD_NAME, indexableText);
+
+        return index;
```

Comment:
❓ Aren't we returning an index that was closed?

Reply:
RAMDirectory's javadoc says that the resource can be left open. I'm thinking now if that's the case then we probably don't need the `try` block for index.